File: src/domain/dto/record.dto.ts
import type { AccountInfoDto } from "./account_info.dto";
import type { CategoryDto } from "./category.dto";

export type RecordDto = {
  id: string;
  type: string;
  amount: number;
  currency: string | undefined | null;
  timestamp: number;
  category: CategoryDto | null | undefined;
  account: AccountInfoDto | null | undefined;
  to_account: AccountInfoDto | null | undefined;
  description: string | null | undefined;
};

File: src-tauri/src/dto/record_dto.rs
use serde::{Deserialize, Serialize};
use super::{account_info_dto::AccountInfoDto, category_dto::CategoryDto};

#[derive(Debug, Serialize, Deserialize)]
pub struct RecordDto {
    pub id: String,
    pub r#type: String,
    pub amount: f64,
    pub currency: Option<String>,
    pub timestamp: i64,
    pub description: Option<String>,
    pub account: Option<AccountInfoDto>,
    pub to_account: Option<AccountInfoDto>,
    pub category: Option<CategoryDto>,
}
